### PR TITLE
added notifications in the workspace sidenav

### PR DESCRIPTION
**Pull request by KaroFr**

### CODE REVIEW

**Comment by KaroFr** [comment]
[Notifications-Komponente](https://trello.com/c/49hSpDSZ/26-notifications-komponente)

File: projects/wave-core-new/src/lib/project/notifications/notifications.component.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+export class NotificationsComponent implements OnInit {
+    notifications: Array<Notification>;
+
+    constructor(private notificationService: NotificationService) {
+        this.notifications = [];
+    }
+
+    ngOnInit(): void {
+        this.notifications = this.notificationService.notifications;
+        this.notifications = this.notifications.reverse();
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/project/notifications/notifications.component.ts:20]: this is not a good idea. If I close/collapse the sidebar and open the settings again, the notification order is reversed every time

File: projects/wave-core-new/src/lib/project/notifications/notifications.component.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    removeAllNotifications(): void {
+        this.notifications = [];
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/project/notifications/notifications.component.ts:24]: this only clears the current view of the notifications. If I close and open the settings again the notifications reappear

File: projects/wave-core-new/src/lib/wave-core.module.ts [hunk 3]
```diff
@@ -215,7 +215,7 @@ const WAVE_COMPONENTS = [
 ];
 
 @NgModule({
-    declarations: [...WAVE_PIPES, ...WAVE_COMPONENTS],
+    declarations: [...WAVE_PIPES, ...WAVE_COMPONENTS, NotificationsComponent],
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/wave-core.module.ts:218]: please add your component to the WAVE_COMPONENTS constant

File: projects/wave-core-new/src/lib/project/notifications/notifications.component.html [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<mat-card *ngIf="notifications.length" [style.overflow]="'auto'" [style.height.px]="'300'">
+    <mat-list *ngFor="let notification of notifications" dense>
+        <mat-list-item>
+            <button class="deleteIcon" mat-icon-button (click)="removeCurrentNotification(notification)">
+                <mat-icon>clear</mat-icon>
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/project/notifications/notifications.component.html:5]: please use the trash-can icon we also use for deleting plots like histogram in the plot view

File: projects/wave-core-new/src/lib/project/workspace-settings/workspace-settings.component.html [hunk 5]
```diff
@@ -48,4 +48,6 @@ <h4 mat-line>New</h4>
         <h4 mat-line>Spatial Reference</h4>
         <p mat-line>Change the current spatial reference</p>
     </mat-list-item>
+    <h3 mat-subheader>Notifications</h3>
+    <wave-notifications></wave-notifications>
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/project/workspace-settings/workspace-settings.component.html:52]: please put a button here instead that leads to a new view with only the notifications shown. They can then use up the whole height of the page

**Review by michaelmattig — CHANGES_REQUESTED** [review]
looks good already but I have some suggestions. Please also merge the latest main into your branch

File: projects/wave-core-new/src/lib/project/notifications/notifications.component.ts (hunk 1 above)
Comment (KaroFr) [reply projects/wave-core-new/src/lib/project/notifications/notifications.component.ts]: d27ffd9b23a6bb4f382ea9cea14eb83f3fb9ba10

File: projects/wave-core-new/src/lib/project/notifications/notifications.component.ts (hunk 2 above)
Comment (KaroFr) [reply projects/wave-core-new/src/lib/project/notifications/notifications.component.ts]: d27ffd9b23a6bb4f382ea9cea14eb83f3fb9ba10

File: projects/wave-core-new/src/lib/wave-core.module.ts (hunk 3 above)
Comment (KaroFr) [reply projects/wave-core-new/src/lib/wave-core.module.ts]: d27ffd9b23a6bb4f382ea9cea14eb83f3fb9ba10

File: projects/wave-core-new/src/lib/project/notifications/notifications.component.html (hunk 4 above)
Comment (KaroFr) [reply projects/wave-core-new/src/lib/project/notifications/notifications.component.html]: d27ffd9b23a6bb4f382ea9cea14eb83f3fb9ba10

File: projects/wave-core-new/src/lib/project/workspace-settings/workspace-settings.component.html (hunk 5 above)
Comment (KaroFr) [reply projects/wave-core-new/src/lib/project/workspace-settings/workspace-settings.component.html]: 05d871119a540a3430c3ee6566796ed0a59d1fb6

**Pull request merged**